accepted/future-releases/static-extension-methods/feature-specification.md

```diff
@@ -46,214 +46,205 @@ extension MyFancyList<T> on List<T> {
+- An expression of the form `e1[e2] op= e3` where `e1` is an extension application `E<...>(e)`, is treated as if it was `E<...>(e)[e2] = E<...>(e)[e2] op e3` except that `e` and `e2` are type-inferred and evaluated only once.
 
-If the member is itself generic and has no type parameters supplied, normal static type inference applies again.
+Increment/decrement operations like `++e` and `e--` are equivalent to composite assignments, except that the post-increment/decrement operations evaluate to one of the intermediate values of the computation.
 
-It is *as if* the invocation `times.quickSort` was converted to `MyList<Duration>(times).quickSort()`. The difference is that there is never an actual `MyList` object, and the `this` object inside `quickSort` is just the `times` list itself (although the extension methods apply to that code too).
+- A pre-increment expression of the form `++e1.id` is generally equivalent to  `e1.id += 1`. Similarly for `--e1.id`, `++e1[e2]` and `--e1[e2]`. This applies when `e1` is an extension application too, reducing it to the former case.
+- A post-increment expression of the form `e1.id++` is generally equivalent to `e1.id += 1` (which is `e1.id = e1.id + 1` except that subexpressions of `e1` are not evaluated more than once), but the value of the expression is the value from evaluating `e1.id` before adding `1`. This too, works similarly when `e1` is an extension application:
+  - `E<...>(e).id++` is equivalent to `E<...>(e).id = E<...>(e).id + 1` except that `e` is only evaluated once, and the value of the increment expression is the value of the subexpression `E<...>(e).id` before the addition. Symmetrically for post-decrement.
+  - `E<...>(e)[e1]++` is equivalent to `E<...>(e)[e2] = E<...>(e)[e2] + 1` except that `e` and `e2` are only evaluated once, and the value of the increment expression is the value of the subexpression `E<...>(e)[e2]` before the addition. Symmetrically for post decrement.
 
-### Generic Parameter Inference
+### Implicit Extension Member Invocation
 
-If both the extension and the method is generic, then inference must infer the extension type parameters first, to figure out whether the extension applies, and only then start inferring method type parameters. As mentioned above, the inference is similar to other cases of chained inference. 
+Extension members can be invoked *implicitly* (without mentioning the extension by name) as if they were members of the `on` type of the extension. This is intended as the primary way to use extensions, with explicit extension member invocation as a fallback for cases where the implicit extension resolution doesn't do what the user want. 
 
-Example:
+An implicit extension member invocation occurs for a simple or composite member invocation with a target expression `e` iff there exists a unique *most specific* extension declaration which is *accessible* and *applicable* to the member invocation (see below).
```

Comment:
I think this leaves the behavior of composite member invocations underspecified (or possibly incorrectly specified) when the setter comes from the extension and the getter from elsewhere (or vice versa).  e.g.

```dart
class A {
  int get x => 3;
  A operator+(int _) => null;
}

extension E on A {
  void set x(int _) {}

}

void main() {
  var a = new A();
  print(a += 3);
}
```

Reply:
Agreed, this comment was written under the understanding that setters and getters were resolved separately, instead of by basename.